editor/scene_tree_dock.cpp

```diff
@@ -3264,6 +3408,31 @@ void SceneTreeDock::_tree_rmb(const Vector2 &p_menu_pos) {
 				}
 			}
 		}
+	} else {
+		if (selection.size() == full_selection.size()) {
+			bool has_internal = false;
+			bool has_top_level = false;
+			bool has_non_editable = false;
+			bool has_non_placeholder = false;
+			for (const Node *node : selection) {
+				if (node) {
+					has_internal |= node->get_scene_file_path().is_empty();
+					has_top_level |= node->get_owner() == nullptr;
+					has_non_editable |= !EditorNode::get_singleton()->get_edited_scene()->is_editable_instance(node);
+					has_non_placeholder |= !node->get_scene_instance_load_placeholder();
+					if (has_internal || has_top_level) {
+						break;
+					}
+				}
+			}
+			if (!has_internal && !has_top_level && profile_allow_editing) {
```

Comment:
`profile_allow_editing` can be checked before going through all the trouble with checking the selection.